src/wikitextprocessor/parser.py

```diff
@@ -541,22 +536,19 @@ def template_parameters(
                                 )
                             continue
 
-                if (is_named and 
-                        (isinstance(parameter_name, int) 
-                        or len(parameter_name)) > 0
-                ):
+                if is_named and len(parameter_name) > 0:  # type: ignore
```

Comment:
I missed the `is_named = False` after the line where parameter_name is cast into int, so this is correct.